### DISCUSSION TITLE
Custom Shields.io badges for Copier repos

### DISCUSSION THREAD

**Discussion by joshuadavidthomas** [copier-org/copier#1516] [Show and tell]
I've been working on two new copier templates, and wanted a way to show off I was using it. So I grabbed the svg logo from `img/copier-logotype.svg`, stripped out the word Copier, and threw it in a public Gist that can then be referenced in a shields.io url. There are even different colors for all the colors in the logo!

Posting this to share, but also gather feedback because I wasn't *totally* sure if I was allowed to do this -- you know, trademarks and everything.

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/b715ed46ac32ddc6f2cbb8e878e2b55752a6bf87/badge.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/b715ed46ac32ddc6f2cbb8e878e2b55752a6bf87/badge-yellow.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/b715ed46ac32ddc6f2cbb8e878e2b55752a6bf87/badge-orange.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/b715ed46ac32ddc6f2cbb8e878e2b55752a6bf87/badge-red.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/b715ed46ac32ddc6f2cbb8e878e2b55752a6bf87/badge-purple.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/b715ed46ac32ddc6f2cbb8e878e2b55752a6bf87/badge-teal.json)

[Link to Gist](https://gist.github.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2)

**Comment by sisp** [comment]
Very cool 🤩, thanks for sharing these badges! 👍 

How about a variant where the logo background is black?

And just another thought: shields.io uses [Simple Icons](https://simpleicons.org/) for their built-in icons, which are monochromatic. How about creating a monochromatic variant of Copier's logo for the badge?

We could add the logo-only SVG and shields.io JSON file to our repo and declare one of these badges official for supporting advertisement that a project or template uses Copier.

> [...]  I wasn't _totally_ sure if I was allowed to do this -- you know, trademarks and everything.

@yajo Do you know if there's any legal issue?

**Reply by joshuadavidthomas** [reply]
> How about a variant where the logo background is black?

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/026975088968fa5e12907ada409db936321c8fc6/black-badge.json)

> And just another thought: shields.io uses [Simple Icons](https://simpleicons.org/) for their built-in icons, which are monochromatic. How about creating a monochromatic variant of Copier's logo for the badge?

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/5f9e888d8b59b1ee08964e2c01cf42e601d079b2/badge-grayscale.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/11227088396f6d95ba30a0a2d9322a305a6054c0/badge-grayscale-inverted.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/67d6cfca962aafe11fd6fa60938417903351aadb/badge-grayscale-border.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/a9831bc40cc69fe5abb93acd84a3e8c08b6ac47e/badge-grayscale-inverted-border.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/026975088968fa5e12907ada409db936321c8fc6/black-badge-grayscale.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/026975088968fa5e12907ada409db936321c8fc6/black-badge-grayscale-inverted.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/026975088968fa5e12907ada409db936321c8fc6/black-badge-grayscale-border.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/026975088968fa5e12907ada409db936321c8fc6/black-badge-grayscale-inverted-border.json)

> We could add the logo-only SVG and shields.io JSON file to our repo and declare one of these badges official for supporting advertisement that a project or template uses Copier.

Totally! I started to open a PR but a) didn't know where to throw the JSON badges and b) wasn't sure if the org would be up for it without discussing it first. Also if it's in the repo, then I imagine all the potential legal issues would go away.

**Reply by joshuadavidthomas** [reply]
![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/7a5bc6b21b30c4f2dfcdcc503e277151fa33591a/black-badge-yellow.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/0108f0bfa8777dd1866c386b1a5b1ccbdf14d963/black-badge-orange.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/7a5bc6b21b30c4f2dfcdcc503e277151fa33591a/black-badge-red.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/7a5bc6b21b30c4f2dfcdcc503e277151fa33591a/black-badge-purple.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/7a5bc6b21b30c4f2dfcdcc503e277151fa33591a/black-badge-teal.json)

**Reply by joshuadavidthomas** [reply]
![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/9aa7e0a67123314959470435bfbbde2e4a4a0312/badge-grayscale-yellow.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/9aa7e0a67123314959470435bfbbde2e4a4a0312/badge-grayscale-orange.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/9aa7e0a67123314959470435bfbbde2e4a4a0312/badge-grayscale-red.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/9aa7e0a67123314959470435bfbbde2e4a4a0312/badge-grayscale-purple.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/9aa7e0a67123314959470435bfbbde2e4a4a0312/badge-grayscale-teal.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/5f15580fc7e06d30f9d8ab7d671e8300a967c1d7/badge-grayscale-inverted-border-yellow.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/9aa7e0a67123314959470435bfbbde2e4a4a0312/badge-grayscale-inverted-border-orange.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/5f15580fc7e06d30f9d8ab7d671e8300a967c1d7/badge-grayscale-inverted-border-red.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/5f15580fc7e06d30f9d8ab7d671e8300a967c1d7/badge-grayscale-inverted-border-purple.json)

![Copier](https://img.shields.io/endpoint?url=https://gist.githubusercontent.com/joshuadavidthomas/7c88611504b557ff7aa2a7524ad996e2/raw/5f15580fc7e06d30f9d8ab7d671e8300a967c1d7/badge-grayscale-inverted-border-teal.json)

**Reply by joshuadavidthomas** [reply]
Okay, I actually need to get some work done 😄 -- you get the idea.

**Comment by yajo** [comment]
FWIW I just recently asked a designer to help with a new logo. It's still a secret, tell nobody! 🕵🏼‍♂️ 

Other than that, I love this initiative, thanks! Is there some way we could provide the needed files directly from this repo? This way, if we happen to have a new logo, those badges would be updated automatically.

**Reply by sisp** [reply]
> FWIW I just recently asked a designer to help with a new logo. It's still a secret, tell nobody! 🕵🏼‍♂️

😲 🤩 

> Is there some way we could provide the needed files directly from this repo?

Absolutely, see e.g. https://github.com/astral-sh/ruff/tree/0c84fbb6db646550e00eed5560a692f2ea1b6224/assets/badge + https://github.com/astral-sh/ruff/blob/0c84fbb6db646550e00eed5560a692f2ea1b6224/README.md?plain=1#L5.
